src/gui/Layout.cpp

```diff
@@ -62,7 +62,7 @@ void Layout::updateVisibility() {
     int y1 = 0;
 
     // Data to select page based on visibility
-    size_t mostPageNr = 0;
+    size_t mostPageNr = -1;
```

Comment:
please use something like `std::numeric_limits<size_t>::max()` and use a predefined constant instead of `-1` (which is also a negative number for unsigned values).
```suggestion
    size_t mostPageNr = INVALID_PAGE_IDX;
```
and define `INVALID_PAGE_IDX` as `constexpr auto INVALID_PAGE_IDX = std::numeric_limits<size_t>::max()` on the top of the source file.

You could also use an std::optional<size_t>, which is a way more readable and only uses 8 byte more on the stack.

Reply:
Thank you, I didn't know about std::optional. I rewrote the patch using it.